lib/cloudregister/registerutils.py

```diff
@@ -160,7 +160,7 @@ def credentials_files_are_equal(repo_credentials):
     """Compare the base credentials files the the repo header and make
        sure they have the same values."""
     credentials_location = '/etc/zypp/credentials.d/'
-    if repo_credentials is None:
+    if not isinstance(repo_credentials, str):
```

Comment:
We also need to return `False` for en empty string. this needs to be `if not repo_credentials or not isinstance(repo_credentials, str)`